tests/subsys/bluetooth/mesh/scheduler_model/message_validity/src/main.c

```diff
@@ -0,0 +1,477 @@
+
+	if (!BT_MESH_ADDR_IS_UNICAST(addr)) {
+		return NULL;
+	}
+
+	index = addr - mock_comp.elem[0].addr;
+	if (index >= mock_comp.elem_count) {
+		return NULL;
+	}
+	return &mock_comp.elem[index];
+}
+
+struct bt_mesh_model *bt_mesh_model_find(const struct bt_mesh_elem *elem, uint16_t id)
+{
+	uint8_t i;
+
+	for (i = 0U; i < elem->model_count; i++) {
+		if (elem->models[i].id == id) {
+			return &elem->models[i];
+		}
+	}
+	return NULL;
+}
+
+int bt_mesh_scene_srv_set(struct bt_mesh_scene_srv *srv, uint16_t scene,
+			  struct bt_mesh_model_transition *transition)
+{
+	return 0;
+}
+
+struct tm *bt_mesh_time_srv_localtime(struct bt_mesh_time_srv *srv, int64_t uptime)
+{
+	static struct tm timeptr;
+	struct bt_mesh_time_tai tai;
+	int64_t tmp = tai_to_ms(&tai);
+
+	tmp += uptime;
+	tai = tai_at(tmp);
+	tai_to_ts(&tai, &timeptr);
+
+	return &timeptr;
+}
+
+void bt_mesh_model_msg_init(struct net_buf_simple *msg, uint32_t opcode)
+{
+}
+
+int64_t bt_mesh_time_srv_mktime(struct bt_mesh_time_srv *srv, struct tm *timeptr)
+{
+	return (timeutil_timegm64(timeptr) - timeutil_timegm64(&start_tm)) * 1000;
+}
+
+int bt_mesh_scene_srv_pub(struct bt_mesh_scene_srv *srv, struct bt_mesh_msg_ctx *ctx)
+{
+	return 0;
+}
+
+int bt_mesh_onoff_srv_pub(struct bt_mesh_onoff_srv *srv, struct bt_mesh_msg_ctx *ctx,
+			  const struct bt_mesh_onoff_status *status)
+{
+	return 0;
+}
+
+void bt_mesh_time_encode_time_params(struct net_buf_simple *buf,
+				     const struct bt_mesh_time_status *status)
+{
+}
+
+int bt_mesh_msg_send(struct bt_mesh_model *model, struct bt_mesh_msg_ctx *ctx,
+		     struct net_buf_simple *buf)
+{
+	if (model->id == BT_MESH_MODEL_ID_SCHEDULER_SRV) {
+		k_sem_give(&status_send);
+	}
+
+	return 0;
+}
+
+int bt_mesh_msg_ackd_send(struct bt_mesh_model *model, struct bt_mesh_msg_ctx *ctx,
+			  struct net_buf_simple *buf, const struct bt_mesh_msg_rsp_ctx *rsp)
+{
+	return 0;
+}
+
+int _bt_mesh_time_srv_update_handler(struct bt_mesh_model *model)
+{
+	return 0;
+}
+
+bool bt_mesh_msg_ack_ctx_match(const struct bt_mesh_msg_ack_ctx *ack, uint32_t op, uint16_t addr,
+			       void **user_data)
+{
+	return false;
+}
+
+/** Call server SET opcode handler */
+static int action_set(const struct bt_mesh_schedule_entry *test_entry,
+		      struct bt_mesh_model *sched_mod, bool ack)
+{
+	BT_MESH_MODEL_BUF_DEFINE(
+		buf, ack ? BT_MESH_SCHEDULER_OP_ACTION_SET : BT_MESH_SCHEDULER_OP_ACTION_SET_UNACK,
+		BT_MESH_SCHEDULER_MSG_LEN_ACTION_SET);
+
+	scheduler_action_pack(&buf, 0, test_entry);
+
+	return _bt_mesh_scheduler_setup_srv_op[ack ? 0 : 1].func(sched_mod, NULL, &buf);
+}
+
+/** Call server GET opcode handler */
+static int current_register_get(struct bt_mesh_model *sched_mod)
+{
+	BT_MESH_MODEL_BUF_DEFINE(buf, BT_MESH_SCHEDULER_OP_GET, BT_MESH_SCHEDULER_MSG_LEN_GET);
+
+	return _bt_mesh_scheduler_srv_op[0].func(sched_mod, NULL, &buf);
+}
+
+/** Call server ACTION GET opcode handler */
+static int action_get(struct bt_mesh_model *sched_mod, uint8_t test_idx)
```

Comment:
All of these take a model pointer as argument, they are also always called with the exact same pointer. In this limited test environment, which model pointer is the server is well-known so I think I would rather just have these functions use that pointer directly instead of having to have the same pointer passed around a lot.